bluemira/geometry/tools.py

```diff
@@ -758,6 +758,73 @@ def sweep_shape(profiles, path, solid=True, frenet=True, label=""):
     return convert(result, label=label)
 
 
+def fillet_chamfer_decorator(chamfer: bool):
+    """
+    Decorator for fillet and chamger operations, checking for validity of wire
+    and radius.
+    """
+
+    def decorator(func):
+        def wrapper(wire, radius):
```

Comment:
We get all the typing and return types properly and the doctrings get copied over as well if we do this

```suggestion
        @functools.wraps(func)
        def wrapper(wire, radius):
```